backend/scrapers/em2.js

```diff
@@ -0,0 +1,371 @@
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    }); 
+    return {
+      event,
+      reason: reasons.reverse().join(', '),
+      success: true,
+    };
+  }
+
+  const goodTermsRes = await this.hasGoodTerms(event);
+  reasons.push(goodTermsRes.reason);
+  if (goodTermsRes.success) {
+    this.skipFurtherChecks.push(event.title);
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });
+    return goodTermsRes;
+  }
+
+
+  this.talkToDB({
+    type: 'db-request',
+    subtype: 'saveRefusedTitle',
+    messageData: {
+      string: event.title,
+      reason: reasons.reverse().join(', '),
+    },
+  });   
+  return {
+    event,
+    reason: reasons.reverse().join(', '),
+    success: false,
+  };
+};
+// #endregion                          MAIN PAGE EVENT CHECK
+
+// #region [rgba(0, 180, 0, 0.1)]      SINGLE PAGE EVENT CHECK
+// #endregion                          SINGLE PAGE EVENT CHECK
+
+// #region [rgba(0, 240, 0, 0.1)]      MAIN PAGE
+scraper.mainPage = async function () {
+  const availableBaseEvents = await this.checkBaseEventAvailable(workerData.family);
+  if (availableBaseEvents) {
+    const thisWorkersEvents = availableBaseEvents.filter(
+      (eventEl, index) => index % workerData.workerCount === workerData.index,
+    );
+    return this.mainPageEnd({ stopFunctie: null, rawEvents: thisWorkersEvents });
+  }
+
+  const { stopFunctie, page } = await this.mainPageStart();
+
+  await this.waitTime(50);
+
+  const cookiesNodig = await page.evaluate(() =>
+    document.querySelector('html').classList.contains('has-open-cookie'),
+  );
+
+  if (cookiesNodig) {
+    await page.evaluate(() => {
+      document.querySelector("[name*='marketing']").click();
+      document.querySelector('.cookie__settings .cookie__process').click();
+    });
+    await this.waitTime(50);
+  }
+
+  await this.autoScroll(page);
+  await this.autoScroll(page);
+
+  let rawEvents = await page.evaluate(
+    // eslint-disable-next-line no-shadow
+    ({ workerData, unavailabiltyTerms }) =>
+      Array.from(document.querySelectorAll('.agenda-container .agenda-item'))
+        .filter((eventEl) => !eventEl.classList.contains('is-hidden'))
+        .map((eventEl) => {
+          const title = eventEl.querySelector('h3')?.textContent ?? '';
+          const res = {
+            anker: `<a class='page-info' href="${document.location.href}">${workerData.family} - main - ${title}</a>`,
+            errors: [],
+            title,
+          };
+
+          try {
+            const DDslashMM = eventEl.querySelector('.agenda-item-date')?.textContent ?? null;
+            const DD = DDslashMM.split("/")[0].trim();
+            const MM = DDslashMM.split("/")[1].trim();
+            res.startDate = `2024-${MM}-${DD}`;
```

Comment:
Dit gaat de soep in bij jaarwissels. Dit doe ik nu inmiddels dus ook anders, maar wat je doet lijkt op een eerder gebruikte logica. In de scraper voor bibelot is er ook niet duidelijk een jaar aanwezig. De aanname die we kunnen doen is dat het eerstkomende event plaats vind in het huidige jaar, stel dat we in april zitten, als we dan in toekomstige events de maand februari zien dan weten we dat er een jaar opgeteld is. Excuseer voor de slordige regexes hier

      const infoElText = document.querySelector('.information')?.textContent.replaceAll(/\s{1,500}/g, ' ') ?? null;
      if (!infoElText) {
        res.errors.push({
          error: new Error('geen infoElText'),
          toDebug: document.querySelector('.information')?.innerHTML ?? 'geen information el',
        });
        return res;
      }
      const datumZonderJaar = Array.isArray(infoElText.match(/datum.*\d{1,2}\s\w{2,4}\s/i)) ? infoElText.match(/datum.*\d{1,2}\s\w{2,4}\s/i)[0].match(/\d{1,2}\s\w+/)[0] : null;
      const maandNaam = datumZonderJaar.match(/[\D]+/i)[0].trim();
      const maandNummer = months[maandNaam];
      const dag = datumZonderJaar.match(/\d+/i)[0].trim().padStart(2, '0');
      const huiMaandNr = (new Date()).getMonth() + 1;
      const huiJaar = (new Date()).getFullYear();
      const jaar = huiMaandNr < Number(maandNummer) ? (huiJaar + 1) : huiJaar;

      res.baseDate = `${jaar}-${maandNummer}-${dag}`;
      if (infoElText.match(/deuren.*\d{1,2}:\d\d\s/i)) {
        res.doorTime = infoElText.match(/deuren.*\d{1,2}:\d\d\s/i)[0].match(/\d\d:\d\d/)[0];
        res.door = `${jaar}-${huiMaandNr}-${dag}T${res.doorTime}:00`;
      }
      res.startTime = infoElText.match(/aanvang.*\d{1,2}:\d\d\s/i)[0].match(/\d\d:\d\d/)[0];
      res.start = `${jaar}-${huiMaandNr}-${dag}T${res.startTime}:00`;